src/components/fields/number-field/number-field.spec.js

```diff
@@ -1,172 +1,210 @@
 import React from 'react';
-import { shallow } from 'enzyme';
+import PropTypes from 'prop-types';
+import omit from 'lodash.omit';
+import { render, fireEvent } from '../../../test-utils';
 import NumberField from './number-field';
-import FieldLabel from '../../field-label';
-import NumberInput from '../../inputs/number-input';
-import FieldErrors from '../../field-errors';
-import { AddBoldIcon } from '../../icons';
-
-const createTestProps = customProps => ({
-  title: 'Username',
-  value: '',
-  onChange: () => jest.fn(),
-  ...customProps,
-});
-
-describe('rendering', () => {
-  describe('data attributes', () => {
-    let textInput;
-    beforeEach(() => {
-      const props = createTestProps({
-        'data-foo': 'bar',
-        'data-test': 'baz',
-      });
-      const wrapper = shallow(<NumberField {...props} />);
-      textInput = wrapper.find(NumberInput);
-    });
-    it('should forward the attributes to the NumberInput', () => {
-      expect(textInput).toHaveProp('data-foo', 'bar');
-      expect(textInput).toHaveProp('data-test', 'baz');
-    });
+
+// This component is used to enable easy testing.
+// It overwrites the onChange function and places a label for the
+// input component. It also ensures an id so that the label can associate
+// the input. This allows tests to use getByLabelText.
+// It also makes sure the event's value passed to onChange flows back to the
+// component so that we can test it under real conditions.
+// As a convenience, we enable accessing a mocked onChange function.
+class Story extends React.Component {
+  static displayName = 'Story';
+  static propTypes = {
+    onEvent: PropTypes.func.isRequired,
```

Comment:
The comment above currently doesn't match the actual behaviour as the `onChange` function does not get overwritten. I would also be careful about introducing `onEvent` as it makes it seem as-if `onEvent` is a function supported by `NumberField`. If you keep this as `onChange` then you also don't need the `omit` below.

We use this pattern in tests of multiple components so people should be familiar with anyways. 

So could you do

```js
handleChange = event => {
  this.props.onChange(event);
  this.setState({ value: event.target.value });
}

<NumberField
  id={this.props.id}
  value={this.state.value}
  onChange={this.handleChange}
/>
```